src/hooks/evm_hook_spec.rs

```
@@ -0,0 +1,71 @@
+use hedera_proto::services;
+
+use crate::contract::ContractId;
+use crate::{
+    FromProtobuf,
+    ToProtobuf,
+};
+
+/// Shared specifications for an EVM hook. May be used for any extension point.
+#[derive(Debug, Clone, PartialEq, Eq)]
+pub struct EvmHookSpec {
+    /// The id of a contract that implements the extension point API with EVM bytecode.
+    pub contract_id: Option<ContractId>,
+}
+
+impl EvmHookSpec {
+    /// Create a new `EvmHookSpec`.
+    pub fn new(contract_id: Option<ContractId>) -> Self {
+        Self { contract_id }
+    }
+}
+
+impl ToProtobuf for EvmHookSpec {
+    type Protobuf = services::EvmHookSpec;
+
+    fn to_protobuf(&self) -> Self::Protobuf {
+        services::EvmHookSpec {
+            bytecode_source: self
+                .contract_id
+                .as_ref()
+                .map(|id| services::evm_hook_spec::BytecodeSource::ContractId(id.to_protobuf())),
```

Are there other variants in `BytecodeSource` in the proto? If the proto adds a new variant later, this code may silently lose information by ignoring it.